src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2826,33 +2826,29 @@ protected function asDateTime($value)
         // these checks since they will be a waste of time, and hinder performance
         // when checking the field. We will just return the DateTime right away.
         if ($value instanceof DateTime) {
```

Comment:
I agree it's not the most efficient approach, but it's what this method does before my PR anyway and results in one less if statement. I doubt it would have a meaningful impact on performance but yeah, it could be added.